compiler/modules/front/codeGen.bal

```diff
@@ -1580,20 +1584,37 @@ function operandIsNil(bir:Operand operand) returns boolean {
 
 final readonly & bir:UniformOrderType[] UNIFORM_ORDER_TYPES = [t:UT_BOOLEAN, t:UT_INT, t:UT_FLOAT, t:UT_STRING];
 
-function operandOrderType(bir:Operand operand) returns bir:OrderType? {
+function operandOrderType(CodeGenContext cx, bir:Operand operand) returns bir:OrderType? {
     if operand is bir:Register {
         t:SemType operandTy = operand.semType;
         if operandTy === t:NIL {
             return ();
         }
-        foreach bir:UniformOrderType tc in UNIFORM_ORDER_TYPES {
-            t:UniformTypeBitSet optBasicType = t:uniformTypeUnion((1 << tc) | (1 << t:UT_NIL));
-            if t:isSubtypeSimple(operandTy, optBasicType) {
-                t:UniformTypeBitSet basicType = t:uniformType(tc);
-                if t:isSubtypeSimple(operandTy, basicType) {
-                    return tc;
+        t:UniformTypeBitSet arrTy = t:uniformTypeUnion(( 1 << t:UT_LIST_RW) | (1 << t:UT_LIST_RO));
+        if t:isSubtypeSimple(operandTy, arrTy) {
+            t:UniformTypeBitSet? memberTy = t:simpleArrayMemberType(cx.mod.env, operandTy, true);
+            if memberTy is t:UniformTypeBitSet {
+                foreach bir:UniformOrderType tc in UNIFORM_ORDER_TYPES {
+                    t:UniformTypeBitSet basicType = t:uniformType(tc);
+                    if (basicType & ~memberTy) == 0 {
+                        return <bir:ArrayOrderType> [{opt:tc}];
+                    }
+                }
+            }
+            else {
+                panic err:impossible("Failed to get array member type");
+            }
+        }
+        else {
+            foreach bir:UniformOrderType tc in UNIFORM_ORDER_TYPES {
```

Comment:
This part should be factored out into a function and reused to get an OrderType for the member of the array.

Reply:
fixed